### PR TITLE
docs: Update messaging Example with null safety

### PR DESCRIPTION
**Pull request by j-j-gajjar**
## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change.*

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by russellwheatley — APPROVED** [review]
There are null check if statements, but I guess it wouldn't harm to explicitly reference the argument nullability. LGTM.

File: packages/firebase_messaging/firebase_messaging/example/lib/main.dart [hunk 1]
```diff
@@ -117,16 +117,16 @@ class _Application extends State<Application> {
...
       if (message != null) {
         Navigator.pushNamed(context, '/message',
             arguments: MessageArguments(message, true));
       }
     });
 
     FirebaseMessaging.onMessage.listen((RemoteMessage message) {
-      RemoteNotification notification = message.notification;
-      AndroidNotification android = message.notification?.android;
+      RemoteNotification? notification = message.notification;
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging/example/lib/main.dart:128]: This is not valid because the file still starts with `// @dart=2.9`

**Review by rrousselGit — CHANGES_REQUESTED** [review]
It appears that the file isn't fully migrated yet.

**Comment by Salakar** [comment]
Closing this due to long period of inactivity and it being incomplete and with conflicts.

@rrousselGit maybe we should just do this ourselves in our own PRs for all the examples anyway?

**Pull request closed without merging**